rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedDependency.java

```diff
@@ -76,16 +85,18 @@ public String getVersion() {
     }
 
     public String getType() {
+        if (type != null) {
+            return type;
+        }
         return requested.getType() == null ? "jar" : requested.getType();
     }
 
-    @Nullable
-    public String getClassifier() {
-        return requested.getClassifier();
+    public @Nullable String getClassifier() {
+        return classifier != null ? classifier : requested.getClassifier();
     }
 
     public boolean isOptional() {
-        return requested.isOptional();
+        return optional != null ? optional : Boolean.valueOf(requested.getOptional());
```

Comment:
Shouldn't this be `resolvedPom.getValue(requested.getOptional())` or something like that? Because the requested form without any resolved context is still a string value.

Reply:
there is no reference to the resovledPom. Again, this code is here to maintain compatibility with LSTs that were serialized prior to this change.